integrate/simpsons.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import "sort"
+
+// Simpsons estimates the integral of a function f
+//  \int_a^b f(x) dx
+// from a set of evaluations of the function using the extended Simpson's rule.
+// The extended Simpson's rule
+// fits a polynomial approximation to the function given
+// two or more adjacent Simpson's rule calculations with shared endpoints,
+// by taking the mean of composite Simpson's 1/8 rule
+// and composite Simpson's 3/8 rule applied to overlapping segments
+// to estimate
+//  \int_x[1]^x[n] f(x) dx
+// where f[i] is the value of the function at x[i].
+// The form and mathematical formulae of extended Simpson's rule can be found at:
+// https://en.wikipedia.org/wiki/Simpson's_rule
+//
+// The (x,f) input data points must be sorted along x.
+// One can use github.com/gonum/stat.SortWeighted to do that.
+// The x and f slices must be of equal length and have length > 1.
+func Simpsons(x, f []float64) float64 {
+	validateInterval(x, f)
+
+	var coefficient, integral float64
+
+	xInitial := x[0]
+	xFinal := x[len(x)-1]
+
+	for i := 0; i < len(x); i++ {
+		switch {
+		case i == 0 || i == len(x)-1:
+			coefficient = 17
+		case i == 2 || i == len(x)-2:
+			coefficient = 59
+		case i == 3 || i == len(x)-3:
+			coefficient = 43
+		case i == 4 || i == len(x)-3:
+			coefficient = 49
+		default:
+			coefficient = 48
+		}
+
+		integral += coefficient * f[i]
```

Comment:
I'm concerned about the numerical accuracy of this approach; sum many potentially large numbers and in the end divide them by something large.